### DISCUSSION TITLE
How to conditionally render components based on `Tabs` component status

### DISCUSSION THREAD

**Discussion by Mil4n0r** [mui/toolpad#3802] [Help and support] [answered]
Hi, I was wondering if there is any known way to render different components based on currently active `Tab` or if there are any plans for adding the possibility to do it.

**Comment by Janpot** [comment, marked as answer]
https://github.com/mui/mui-toolpad/pull/1549 contains some instructions on how this component was intended to be used.
I'm adding [an example](https://github.com/mui/mui-toolpad/pull/3803) to our repository to illustrate the usage better.
I'm also opening a ticket for [improved documentation](https://github.com/mui/mui-toolpad/issues/3804) around this feature.

For an improved UX You can upvote the ticket https://github.com/mui/mui-toolpad/issues/747.

**Comment by Mil4n0r** [comment]
Sorry about revisiting this topic after marking as resolved. I realised that a small margin is added with each container added (even though all of them are hidden except for one), which is specially notable when the Tabs number becomes high. Would it be possible to overcome this problem? The problem can be easily spotted in the video from this [comment](https://github.com/mui/mui-toolpad/issues/747#issue-1330181228) at 1:00

**Reply by Janpot** [reply]
This should have been fixed in https://github.com/mui/mui-toolpad/pull/3803. Which version of Toolpad are you using?

**Reply by Mil4n0r** [reply]
I am using v1.50 and cannot update to latest version because of a problem that I am facing when using components that require styled-components (https://github.com/mui/mui-toolpad/issues/3643).

Is there any workaround that I can do in the current version? Nonetheless I will try to prepare a demo soon to demonstrate the problem stated in that issue. The following line is not fixing it for me:
```
'&:empty': {
  display: 'none',
},
```

**Reply by Mil4n0r** [reply]
It is now solved, the problem was that the Container that I was using had a slightly different Wrapper than yours. In my case I wasn't setting it to null but to `display: none` and that was causing the issue.

I still have a bit of trouble when dragging and dropping in tab containers other than the first, but that seems to happen in latest version as well and it does not seem like a major problem.

Thanks for your help!

**Reply by Janpot** [reply]
Thank you for the feedback, I'm opening https://github.com/mui/mui-toolpad/issues/3865 to hopefully improve the UX of working with container components.
